lib/Dialect/FSM/FSMOps.cpp

```diff
@@ -76,6 +76,8 @@ void MachineOp::getHWPortInfo(SmallVectorImpl<hw::PortInfo> &ports) {
   for (unsigned i = 0, e = machineType.getNumInputs(); i < e; ++i) {
     hw::PortInfo port;
     port.name = getArgName(i);
+    if (!port.name)
+      port.name = StringAttr::get(getContext(), "in" + std::to_string(i));
```

Comment:
```suggestion
      port.name = StringAttr::get(getContext(), "in" + Twine(i))
```
May need to be `Twine("in") + Twine(i)` or something.  No need to allocate a string, concatenate them into a new heap allocation, just to get the final "in5".  Similarly below.

Reply:
twine of an integer allocates a string anyway I believe.

Reply:
Nope, that's not how it works.  It's very clear if you open up `Twine.h` or look at its implementation.   Not doing that is an important reason for it to exist at all (as documented) -- it stashes the integer and then it can all be put into a single final string.

See:

https://github.com/llvm/llvm-project/blob/95ae517bb4c332149b3c45cd7d1e6edf8b40077e/llvm/include/llvm/ADT/Twine.h#L28-L35 .